File: tool/src/main/java/com/glodon/tool/exception/LogFileUtil.java
package com.glodon.tool.exception;

import com.glodon.tool.FileUtil;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;

/**
 * Created by lichongmac@163.com on 2019-12-20.
 */
public class LogFileUtil {

    public static void writeLogFile(String fileName,String msg,String path){

        String filePath = FileUtil.createNewFile(path, fileName);
        FileOutputStream outStream = null;
        try {
            outStream = new FileOutputStream(new File(filePath),true);
            outStream.write(msg.getBytes("UTF-8"));
            outStream.flush();
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            if (outStream != null) {
                try {
                    outStream.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }
}

File: kotlintools/src/main/java/com/glodon/glm/kotlintools/LogFileUtil.kt
package com.glodon.glm.kotlintools

import com.glodon.glm.kotlintools.FileUtil.createNewFile
import java.io.File
import java.io.FileOutputStream
import java.io.IOException

/**
 * Created by lichongmac@163.com on 2020/6/12.
 */
object LogFileUtil {

    fun writeLogFile(fileName: String?, msg: String, path: String?) {
        val filePath: String = createNewFile(path, fileName)
        var outStream: FileOutputStream? = null
        try {
            outStream = FileOutputStream(File(filePath), true)
            outStream.write(msg.toByteArray(charset("UTF-8")))
            outStream.flush()
        } catch (e: IOException) {
            e.printStackTrace()
        } finally {
            if (outStream != null) {
                try {
                    outStream.close()
                } catch (e: IOException) {
                    e.printStackTrace()
                }
            }
        }
    }
}